contracts/staking/EpochSelector.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./interfaces/IEpochSelector.sol";
+import "./ClusterSelector_wih_abi_encode.sol";
+
+/// @title Contract to select the top 5 clusters in an epoch
+contract EpochSelector is AccessControl, ClusterSelector, IEpochSelector {
+    /// @notice Event emitted when Cluster is selected
+    /// @param epoch Number of Epoch
+    /// @param cluster Address of cluster
+    event ClusterSelected(uint256 indexed epoch, address indexed cluster);
+
+    /// @notice length of epoch
+    uint256 public constant epochLength = 4 hours;
```

Comment:
Tradeoff here is between how many txs need to be submitted by receiver vs how soon can we rotate the clusters so that if they are offline, the network doesn't suffer.

Reply:
The epoch length also should be long enough for someone to send a tx to rotate the clusters selected for the epoch.